tool.ts

```diff
@@ -235,21 +233,42 @@ export async function downloadTool(
 
             tl.debug('downloading');
             let response: httpm.HttpClientResponse = await http.get(url, additionalHeaders);
-            
+
             if (response.message.statusCode != 200) {
                 let err: Error = new Error('Unexpected HTTP response: ' + response.message.statusCode);
                 err['httpStatusCode'] = response.message.statusCode;
                 tl.debug(`Failed to download "${fileName}" from "${url}". Code(${response.message.statusCode}) Message(${response.message.statusMessage})`);
                 throw err;
             }
 
+            let downloadedContentLength = _getContentLengthOfDownloadedFile(response);
+            if (!isNaN(downloadedContentLength)) {
+                tl.debug(`Content-Length of downloaded file: ${downloadedContentLength}`);
+            } else {
+                tl.debug(`Content-Length header missing`);
+            }
+
             tl.debug('creating stream');
             let file: NodeJS.WritableStream = fs.createWriteStream(destPath);
             file.on('open', async (fd) => {
                 try {
                     let stream = response.message.pipe(file);
                     stream.on('close', () => {
                         tl.debug('download complete');
+                        let fileSizeInBytes = _getFileSizeOnDisk(destPath);
+                        if (!isNaN(fileSizeInBytes)) {
+                            tl.debug(`Downloaded file size: ${fileSizeInBytes} bytes`);
+                        } else {
+                            tl.debug(`File size on disk was not found`);
+                        }
+
+                        if (!isNaN(downloadedContentLength) &&
+                            !isNaN(fileSizeInBytes) &&
+                            fileSizeInBytes !== downloadedContentLength) {
+                            let err: Error = new Error(`Content-Length (${downloadedContentLength} bytes) did not match downloaded file size (${fileSizeInBytes} bytes).`);
+                            reject(err);
```

Comment:
Will mismatch between 'Content-Length' and actual file size of downloaded file always means the error? Are there some cases we missed here like gzip we could miss here (see [this ](https://blog.actorsfit.com/a?ID=01750-192152cd-a581-49af-9ecf-3775cb415fc0) for example)?
I'm a little bit worrying also that we don't have ability to switch to the previous behavior - could it make sense to introduce at least some variable to switch?

Reply:
Does it make sense just to write the warning in this case? I agree that in some cases we will fail the task without real reason, when content-length mismatch file size on disk. 
What do you think?

Reply:
@tatyana-kostromskaya hmm, I think this depends on a goal of such check and cases here, and also which cases happens more often...
Have you tested these changes? Does it match for you?
Is it ever possible that these fields would be mismatched?
Can we distinguish between some issues during network transmission and other reasons of this?
If it is still make sense to add such check - should we add ability to disable it - to consider other cases?